LogAnalysis.hs

```diff
@@ -0,0 +1,101 @@
+{-# OPTIONS_GHC -Wall #-}
+module LogAnalysis where
+import Log
+
+-- 'I' is for informational messages 
+-- 'W' for warnings
+-- 'E' for errors 
+
+-- Error messages have scale on 1 to 100 (i.e. mild to extreme)
+
+---------------------------------------------------------------------------------
+-- Exercise 1: parseMessage
+---------------------------------------------------------------------------------
+parseMessage :: String -> LogMessage  -- parsing function 
+
+-- $ means that anything after it will take precedence over anything before it 
+parseMessage s = case words s!!0 of
+    "I" -> LogMessage Info (read $ words s!!1 :: TimeStamp) (unwords $ drop 2 $ words s)
```

Comment:
You're calling `words` function here more than once. This is not efficient in general. Also you're using indexing operators `!!`. This function can be written cleaner using pattern matching. Something like this:

```haskell
parseMessage s = case words s of
    ("I":n:text) ->
    ... 
```

This is more cleaner and more efficient since with this approach you're decomposing program into independent different cases.